File: go/hamming/hamming.go
// Package hamming provides handy method for handle with Hamming difference
package hamming

import "errors"

// Distance calcule the Hamming difference between two DNA strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("lenght of a and b can't be different")
	}
	distance := 0
	for i:= 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil;
}

File: python/hamming/hamming.py
"""
hamming module function to calculate the Hamming Distance between two DNA
strands
"""


def distance(strand_a, strand_b):
    """
    Calculate the Hamming Distance between two DNA strands
    """
    if len(strand_a) != len(strand_b):
        raise ValueError("Invalid strands")
    return len([a for a, b in zip(strand_a, strand_b) if not a == b])
